FBTweak/FBTweak.h

```diff
@@ -48,121 +48,148 @@ typedef id FBTweakValue;
+ @discussion Optional. If nil, there is no minimum. Numeric only.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue stepValue;
+@property (nonatomic, strong, readwrite, nullable) FBTweakValue minimumValue;
 
 /**
-  @abstract The decimal precision value of the tweak.
-  @discussion Optional. If nil, the precision value is calculated from
-    the step value. Only used for numeric tweaks.
+ @abstract The maximum value of the tweak.
+ @discussion Optional. If nil, there is no maximum. Numeric only.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue precisionValue;
+@property (nonatomic, strong, readwrite, nullable) FBTweakValue maximumValue;
 
 /**
-  @abstract Adds an observer to the tweak.
-  @param observer The observer. Must not be nil.
-  @discussion A weak reference is taken on the observer.
+ @abstract The step value of the tweak.
+ @discussion Optional. If nil, the step value is calculated from
+ the miniumum and maxium values. Only used for numeric tweaks.
  */
-- (void)addObserver:(id<FBTweakObserver>)observer;
+@property (nonatomic, strong, readwrite, nullable) FBTweakValue stepValue;
 
 /**
-  @abstract Removes an observer from the tweak.
-  @param observer The observer to remove. Must not be nil.
-  @discussion Optional, removing an observer isn't required.
+ @abstract The decimal precision value of the tweak.
+ @discussion Optional. If nil, the precision value is calculated from
+ the step value. Only used for numeric tweaks.
  */
-- (void)removeObserver:(id<FBTweakObserver>)observer;
+@property (nonatomic, strong, readwrite, nullable) FBTweakValue precisionValue;
 
 @end
 
 /**
-  @abstract Responds to updates when a tweak changes.
+ @abstract An immutable tweak.
  */
-@protocol FBTweakObserver <NSObject>
+@interface FBBaseTweak : NSObject <FBTweak>
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ @abstract Initializes with the given \c name, \c identifier and \c currentValue.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
+                      currentValue:(FBTweakValue)currentValue NS_DESIGNATED_INITIALIZER;
+
+@end
 
 /**
-  @abstract Called when a tweak's value changes.
-  @param tweak The tweak which changed in value.
+ @abstract A tweak with an underlying block.
  */
-- (void)tweakDidChange:(FBTweak *)tweak;
+@interface FBActionTweak : NSObject <FBActionTweak>
 
-@optional
+- (instancetype)init NS_UNAVAILABLE;
 
 /**
- @abstract Called when a tweak's value will change.
- @param tweak The tweak which value will change.
+ @abstract Initializes with \c identifier, \c name and \c block as the underlying block.
+ The \c currentValue property of the tweak is set to be the \c block.
  */
-- (void)tweakWillChange:(FBTweak *)tweak;
+- (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
+                             block:(dispatch_block_t)block NS_DESIGNATED_INITIALIZER;
 
 @end
+
+/**
+  @abstract The default implementation for \c FBEditableTweak.
```

Comment:
I'd remove the `default`.

Reply:
Not sure what you mean, `The implementation for \c FBEditableTweak` doesn't seem right. I use the term (default implementation) a lot.